modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java

```diff
@@ -201,6 +201,7 @@ public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings sett
                     .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
                     .build()
             )
+            .setIndexFormat(new SystemIndexDescriptor.IndexFormat(0, "-776058359"))
```

Comment:
Where are these values got from? Could they be calculated on the fly?

Reply:
I got the values by writing failing unit tests.

It would be easy to get them at runtime. But consider the case where a dev changes mappings or settings, but doesn't bump the system index version number. If we hard-code these version numbers, we'll fail early and the dev can look at documentation (to be written) to see how to correctly increment the version.

If we calculate these values at runtime, then we can potentially detect a problem when a bwc test contains system indices with equal versions but different hash values, but we won't be able to add that assertion until the PR where minimum system index versions are added to DiscoveryNodes.

Reply:
Rather than just having single ints, it would be better if the way the values were derived is explicit - at the start of the test, have a `int hash = Objects.hash(Settings.builder().put(...).build(), ...)` that was calculated from fixed values, so you can see where the hash has come from.